### PR TITLE
fix: autocomplete-regress

### LINKED ISSUE #15859: [Bug]-[10937.5]:Autocomplete won't suggest appsmith.geoLocation.currentLocation until appsmith.geolocation.watchPosition() is run
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I'm not able to use `appsmith.geolocation.currentPosition` on the JS Object editor or on the JS editor on the right pane unless I run `appsmith.geoLocation.currentLocation` at least once. This is confusing as the autocomplete results should be consistent across scenarios. 

### Steps To Reproduce

1. Open the JS Object editor, try typing `appsmith.geolocation.currentPosition` and notice that you won't be able to use it
2. Run `appsmith.geoLocation.currentLocation` once 
3. Try typing `appsmith.geolocation.currentPosition` and you will now be able to successfully use it

https://user-images.githubusercontent.com/17562726/183604553-336dfd04-732c-4cfd-86fc-82e40c753b78.mp4

### Public Sample App

https://app.appsmith.com/app/testbed/page1-62f0b9e9b2f8027edfbf9008

### Version

Cloud

### LINKED ISSUE #16158: [Task]: Add Cypress test for autocompletion of binding in array and object value
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

_No response_

### LINKED ISSUE #16188: [Bug]: Remove autocomplete for groupButton property of ButtonGroupWidget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Remove suggestion for `groupbuttons` property of `ButtonGroupWidget`. 

### Steps To Reproduce

1. Create new Appsmith app
2. Add ButtonGroupWidget
3. Add text widget
4. Type `{{Button`, you could notice autocomplete suggestion for buttonGroupWidget which shouldn’t be shown.

### Public Sample App

_No response_

### Version

1.7.13

### LINKED ISSUE #16288: [Bug]: Auto-complete does not work when multiple bindings exists on the same line
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I try to use a binding in an INSERT statement of Snowflake DB, though the widgets already exist on the page, the Autocomplete does not work to complete the widget names and action. 

### Steps To Reproduce

Create a snowflake datasource and create an INSERT query to the same. 

Refer following video to reproduce:
https://www.loom.com/share/91a7957ccc0b45c295685eb8faf388f2

### Public Sample App

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

Fixes #16288
Fixes #16158
Fixes #16188
Fixes #15859

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- [x] [Add Multiple bindings in single-line JS code and test autocomplete ](https://github.com/appsmithorg/TestSmith/issues/2063)

Addition test
- [x] Create 2 page, add the widget to 1st page then switch to 2nd page and check autocompletion in bindings
For this issue https://github.com/appsmithorg/appsmith/issues/16569

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by ohansFavour — APPROVED** [review]

**Comment by ramsaptami** [comment]
All fixes have been verified on the DP. GTG to release
cc: @Rishabh-Rathod

**Comment by Rishabh-Rathod** [comment]
Adding test plan approved as it is verified.

File: app/client/src/utils/autocomplete/EntityDefinitions.ts [hunk 1]
```diff
@@ -275,13 +274,11 @@ export const entityDefinitions = {
...
     recaptchaToken: "string",
   },
-  BUTTON_GROUP_WIDGET: (widget: ButtonGroupWidgetProps) => {
-    return {
-      "!doc":
-        "The Button group widget represents a set of buttons in a group. Group can have simple buttons or menu buttons with drop-down items.",
-      "!url": "https://docs.appsmith.com/widget-reference/button-group",
-      groupButtons: generateTypeDef(widget.groupButtons),
-    };
+  BUTTON_GROUP_WIDGET: {
```
[the remark this answers is not in the document]
Comment (Rishabh-Rathod) [reply app/client/src/utils/autocomplete/EntityDefinitions.ts:277]: https://github.com/appsmithorg/appsmith/issues/16822

**Pull request merged**